flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/rest/SqlGatewayRestEndpointTest.java

```diff
@@ -0,0 +1,169 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.gateway.rest;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.RestOptions;
+import org.apache.flink.runtime.rest.RestServerEndpointConfiguration;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.gateway.api.endpoint.SqlGatewayEndpointFactoryUtils;
+import org.apache.flink.table.gateway.rest.util.SqlGatewayRestEndpointFactory;
+import org.apache.flink.table.gateway.rest.util.SqlGatewayRestOptions;
+import org.apache.flink.util.ConfigurationException;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.jupiter.api.Test;
+
+import static org.apache.flink.table.gateway.api.endpoint.SqlGatewayEndpointFactoryUtils.getEndpointConfig;
+import static org.apache.flink.table.gateway.rest.SqlGatewayRestEndpointITCase.getBaseConfig;
+import static org.apache.flink.table.gateway.rest.SqlGatewayRestEndpointITCase.getSqlGatewayRestOptionFullName;
+import static org.apache.flink.table.gateway.rest.util.SqlGatewayRestEndpointFactory.IDENTIFIER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/** Tests for the {@link SqlGatewayRestEndpoint}. */
+public class SqlGatewayRestEndpointTest extends TestLogger {
+
+    private static final String ADDRESS1 = "123.123.123.123";
+    private static final String ADDRESS2 = "123.123.123.144";
+    private static final String BIND_ADDRESS1 = "023.023.023.023";
+    private static final String BIND_ADDRESS2 = "023.023.023.333";
+    private static final String BIND_PORT1 = "7282";
+    private static final String BIND_PORT2 = "7444";
+    private static final String PORT1 = "7661";
+    private static final String PORT2 = "7662";
+    public static final String SQL_GATEWAY_ADDRESS =
+            getSqlGatewayRestOptionFullName(SqlGatewayRestOptions.ADDRESS.key());
+    public static final String SQL_GATEWAY_BIND_ADDRESS =
+            getSqlGatewayRestOptionFullName(SqlGatewayRestOptions.BIND_ADDRESS.key());
+    public static final String SQL_GATEWAY_BIND_PORT =
+            getSqlGatewayRestOptionFullName(SqlGatewayRestOptions.BIND_PORT.key());
+    public static final String SQL_GATEWAY_PORT =
+            getSqlGatewayRestOptionFullName(SqlGatewayRestOptions.PORT.key());
+
+    /**
+     * Test {@link SqlGatewayRestEndpoint} uses its own options when there are both runtime options
+     * and sql gateway options in the delegating configuration.
+     */
+    @Test
+    public void testIfSqlGatewayRestEndpointUseOverrideOptions() throws ConfigurationException {
```

Comment:
```suggestion
    void testIfSqlGatewayRestEndpointUseOverrideOptions() throws ConfigurationException {
```
All test method and lifecycle method should be package-private.

Reply:
Done.